### PR TITLE
Feat/2482 ci cache rust artifact

### LINKED ISSUE #2482: CI should cache rust artifact to reduce build time
Cache rust artifact file to reduce the build time.

here is a list of file to cache:

- [ ] cache `target`
- [ ] cache `~/.cargo/git`
- [ ] cache `~/.cargo/registry`

docs [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/release/caching?view=azure-devops)

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
closes #2482

### CODE REVIEW

File: .azure-pipelines.yml
Comment (touilleMan) [inline .azure-pipelines.yml:680]: the `s` is here given it refers to modules that could be not available (so it's not a boolean information)

**Review by touilleMan — COMMENTED** [review]
Do we have an idea how much time we save with this ? 😃

File: .azure-pipelines.yml
Comment (FirelightFlagboy) [reply .azure-pipelines.yml:680]: `available` is an adjective, it don't take a 's' on plurial.

maybe `MODULES_NOT_AVAILABLE` sound better ?

**Comment by FirelightFlagboy** [comment]
> Do we have an idea how much time we save with this ? smiley

on `oxidation` `Rust Test` is now `~1 min` (come from `~6 min`)

File: .azure-pipelines.yml
Comment (FirelightFlagboy) [reply .azure-pipelines.yml]: done in a8ae932

**Comment by touilleMan** [comment]
plz wait for #2371 before merge this ^^

**Comment by TimeEngineer** [comment]
#2371 merged !

**Review by TimeEngineer — APPROVED** [review]

**Pull request merged**
